### ISSUE TITLE
Supported Visual Studio versions

### ISSUE THREAD

**Issue by topeterk** [TwitchLib/TwitchLib#414]
Hi,

I cloned the repository today and tried to open up the solution but it fails on almost every project file because of this:
_"The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format."_

The Solution seems to be built for Visual Studio 2015 but searching for the message on the web points me in using Visual Studio 2017 leading me to the question:
**What Visual Studio version is the source compatible with and is there a chance to get it working with Visual Studio 2015?**

Thank you,
  Peter

PS: Maybe add a note to the Readme about the requirements (if there are any and I read over it)?

**Comment by swiftyspiffy** [comment]
You might want to clone the individual repos, and not the "main" repo as it'll contain gitmodule links. Check the organization, and clone out the api, client and pubsub repos.

**Comment by topeterk** [comment]
Just tried downloading the repositories separately but the failures is still the same.
Also tried the other way of updating everything to master with the cloned "main" repository, means:
> git submodule foreach --recursive git checkout master
> git submodule foreach --recursive git pull
But same result there: the folders are the same now and also the error.

Tried opening all 4 available VS solutions cannot load any project expect Client.Test.

In the meantime I also tried updating Visual Studio 2015 to Update 3 and I also installed the Development Packs for 4.6.2 and 4.7.2 from here:
> https://www.microsoft.com/net/download/visual-studio-sdks

So far nothing helped. Any ideas?

PS: As you haven't answered my first question: Do you actually know that it still works everything with Visual Studio 2015 and what version are you using which is know to work?

**Comment by LuckyNoS7evin** [comment]
All of us are using VS2017+ the main reason for this is it's not a 3 year old bit of software and it also includes the newer package and project management features.

Now I know in some cases you may not be able to run 2017 for other projects however 2015 and 2017 can run side by side and with Community edition being free that is also an option.

TwitchLib is built for dotnet core 2.1 and dotnet standard so is only supported at 4.6.2 and above.

**Comment by LuckyNoS7evin** [comment]
The structure of csproj files changed between 2015 and 2017. It will upgrade will not downgrade

**Comment by topeterk** [comment]
Tried to load precompiled binaries via NuGet in VS2015 but TwichLib requires .net 4.6.2 which seems to be not compatible with Microsoft.Extensions.Logging and I cannot find any working combination.
And since I saw that I have thousands of dependencies and I'm being forces to use latest runtime and development tools, I think I'm lost here.

My software runs with .Net 2.0 and anything onwards and I don't like to depend on so many other libraries, frameworks and licenses just the necessity of my very few IRC calls.

Maybe one day with another project I will try use this library with VS2017/19/++.
Thank you anyway.
